lib/augmentConfig.mjs

```diff
@@ -362,6 +389,52 @@ async function addPluginFunctions(config, { quietDeprecationWarnings }) {
 	return config;
 }
 
+/**
+ *
+ * @param {StylelintConfig} config
+ * @returns {Promise<StylelintConfig>}
+ */
+async function addProcessorFunctions(config) {
+	if (!config.processors) {
+		return config;
+	}
+
+	const processorPromises = config.processors.map(async (processorLookup) => {
+		let processorImport;
+
+		if (typeof processorLookup === 'string') {
+			processorImport = await dynamicImport(processorLookup);
+			processorImport = processorImport.default || processorImport;
+		} else {
+			processorImport = processorLookup;
+		}
```

Comment:
[suggest] Since `config.processors` should be an array of string, how about simplifying the code like this?

```suggestion
		let processor = await dynamicImport(processorLookup);

		processor = processor.default ?? processor;

		if (!isFunction(processor)) {
			throw configurationError(`The processor "${processorLookup}" must be a function`);
		}
```

`isFunction` is available by importing `utils/validateTypes.mjs` as below:

```js
import { isFunction } from './utils/validateTypes.mjs';
```